src/TestSuite/IntegrationTestCase.php

```diff
@@ -445,6 +454,11 @@ protected function _assertStatus($min, $max, $message)
             $this->fail('No response set, cannot assert status code.');
         }
         $status = $this->_response->statusCode();
+
+        if ($this->_exception && ($status < $min || $status > $max)) {
+            throw $this->_exception;
```

Comment:
Shouldn't you fail() instead of throw? Throwing will make the test as an 'error' which is not what actually happened.